### PR TITLE
chore(infra): test Node.js v16 and expand "engines" range to include it

### PR DESCRIPTION
**Pull request by abernix**
This PR introduces testing for and expands package's `engines` range for the `node` engine to support the v16 line, which was recently released and is scheduled to go into it's Long-Term Support contract with the Node.js Foundation in October.  This PR targets the `npm@7` upgrade PR (#708).

I've made a couple other related changes within this PR, but of notable importance, I've intentionally disabled Renovation of Docker images within CircleCI because I'm _pretty sure_ that right now, by having moved the `docker`.`image` property into this file, rather than in the Orb (which I think is the _right_ direction to go), it would just be updating everything to the true numerical latest version (e.g., 16, or even 17), which is not what we want.  (e.g., all our tests which intentionally test different versions of Node.js would test the same _recent_ version).  I think I know how to fix this, but I don't want to couple it to this PR and don't have time to look into it further at the moment (and would benefit from some local development time with Renovate and this PR actually having landed already).

### CODE REVIEW

File: renovate.json5
Comment (abernix) [inline renovate.json5:19]: Note to self: Another option might be https://docs.renovatebot.com/configuration-options/#matchcurrentversion.

File: netlify.toml
Comment (glasser) [inline netlify.toml:11]: I will trust you that this works; I see that the docs explicitly say that NODE_VERSION can be a single number and hope that the same is true for NPM_VERSION.

File: renovate.json5
Comment (glasser) [inline renovate.json5:12]: I'm not trying to verify whether this is true or not (I kinda hope it's actually not hard to tell it not to take major version bumps), happy to move forward with circleci disabled for now even if there's a chance you're being overly conservative.

**Review by glasser — APPROVED** [review]

**Review by martijnwalraven — APPROVED** [review]
I don't know enough about CircleCI and Renovate to review those changes, but the rest looks good to me!

File: renovate.json5
Comment (abernix) [reply renovate.json5:12]: I suspect it's easy to tell it to not take major version bumps for this file, yes.

File: netlify.toml
Comment (abernix) [reply netlify.toml:11]: Taken from the Netlify build logs from this branch:

```
3:28:00 PM: v14.16.1 is already installed.
3:28:00 PM: Now using node v14.16.1 (npm v6.14.12)
3:28:03 PM: Found npm version (6.14.12) that doesn't match expected (7)
Installing npm at version 7
3:28:10 PM: /opt/buildhome/.nvm/versions/node/v14.16.1/bin/npx -> /opt/buildhome/.nvm/versions/node/v14.16.1/lib/node_modules/npm/bin/npx-cli.js
3:28:10 PM: /opt/buildhome/.nvm/versions/node/v14.16.1/bin/npm -> /opt/buildhome/.nvm/versions/node/v14.16.1/lib/node_modules/npm/bin/npm-cli.js
3:28:10 PM: + npm@7.11.2
```

**Comment by abernix** [comment]
Just FYI: I landed https://github.com/apollographql/federation/pull/713/commits/e25349b2be671ae162865cf9295ccea8325fe85f via #729, rather than packaging it within this PR.

**Comment by holm** [comment]
I'm trying to update to Node 17 in my project, but I am blocked by yarn, as these packages restrict the use of this engine. It seems very pessimistic to have an upper bound on the node version, as for the vast majority of times, a new Node version will not break these packages. This can also be seen by there being no actually code change in this PR.

**Comment by holm** [comment]
For reference, for the gateway package, the upper limit was added in this commit: https://github.com/apollographql/federation/commit/4ef79ae8a5034dfcc8fef0fe4400d6fee0f5a259#diff-e441a2bec2f27de2e89f69427e4837410164f9ad4dd7e9428e4286b024ee5d78R15

But I can't really find anything in the commit message or PR that indicates why this was done.

**Comment by trevor-scheer** [comment]
@holm It’s not unreasonable for a library to be pessimistic about supporting various major versions without first validating that things aren’t broken. No upper bound means we're claiming support for ALL future major versions of Node in those releases. We'd rather ship with a clear expectation of what we support than allow our users to break unexpectedly. Yarn supports working around this (buyer beware) via the `--ignore-engines` flag, which I welcome you to use! I’d also welcome you to open a separate issue or PR to introduce testing and support for Node v17.

**Comment by holm** [comment]
It may not be unreasonable, but I don't think it's very helpful. NodeJS goes out of their way to be as backwards compatible as possible and from my experience, it is incredibly rare for packages to break in new node versions, unless they rely on internal APIs og uses C code.

`--ignore-engines` is painful to use. You cannot persist the config, so  your CI system, your deployment system, etc has to be updated. In addition all collaborators now have to remember to add the flag every time they run yarn to install new packages, which is a daily occurrence. It also has the big drawback that it will ignore cases where your engine version is too low, which is really what the yarn check is all about, see. f.ex. https://github.com/yarnpkg/rfcs/pull/69#issuecomment-308567694 directly from `Saccone`.

It's the choice of the maintainers of course, I think it's just an annoyance with a tiny benefit.

**Pull request merged**
